app/blog/2018-03-14-polymer-3-0.md

```diff
@@ -0,0 +1,330 @@
+```js
+import '@polymer/polymer/lib/elements/dom-if.js';
+```
+
+To import a Polymer Element:
+```js
+import '@polymer/paper-checkbox/paper-checkbox.js';
+```
+
+To import a behavior:
+```js
+import {IronResizableBehavior}
+    from '@polymer/iron-resizable-behavior/iron-resizable-behavior.js';
+```
+
+To import utility modules like `async` that export several members, you can import individual members, or import the entire module:
+
+```js
+import * as async from '@polymer/polymer/lib/utils/async.js'
+async.microTask.run(callback);
+```
+
+The Polymer 3.0 API docs are still in progress-so for the moment, refer to the [Polymer 3.0 preview source code](https://github.com/Polymer/polymer/tree/__auto_generated_3.0_preview) to determine what a module exports.
+
+#### Old Polymer 3.0 preview syntax
+
+```js
+/* my-app.js
+ * 
+ * Old syntax used path names.
+ * Also, the main export of polymer-element.js
+ * was called Element and had to be renamed on import. 
+ */
+import { Element as PolymerElement } from './node_modules/@polymer/polymer/polymer-element.js';
+import './node_modules/@polymer/polymer/lib/elements/dom-if.js';
+import './node_modules/@polymer/paper-checkbox/paper-checkbox.js';
+import './my-view.js';
+
+class MyApp extends PolymerElement {
+  static get template () {
+    // Old syntax let you return a string literal.
+    return `
+      <h1>Hello World</h1>
+      <template is="dom-if" if="true">
+        <paper-checkbox>I like pie</paper-checkbox>
+      </template>
+      <my-view></my-view>
+    `;
+  }
+}
+customElements.define('my-app', MyApp);
+```
+
+```html
+<!-- index.html -->
+<!-- this doesn't change -->
+<!-- polyfills, metadata, etc -->
+<script type="module" src="./my-app.js">
+<my-app></my-app>
+```
+
+#### Current Polymer 3.0 preview syntax
+
+```js
+/* 
+ * my-app.js
+ * 3 changes:
+ * - Import statements now use package names instead of paths
+ * - polymer-element.js exports PolymerElement instead of Element, no 
+ *   need to rename it
+ * - Import the html helper function from polymer-element.js for templates
+*/
+import { PolymerElement, html } from '@polymer/polymer/polymer-element';
+import '@polymer/polymer/lib/elements/dom-if';
+import '@polymer/paper-checkbox/paper-checkbox';
+// Continue to use paths for importing other modules that are part of your app
+import './my-view.js';
+
+class MyApp extends PolymerElement {
+  static get template () {
+    // Always return an HTMLTemplateElement, not a string literal
+    return html`
+      <h1>Hello World</h1>
+      <template is="dom-if" if="true">
+        <paper-checkbox>I like pie</paper-checkbox>
+      </template>
+      <my-view></my-view>
+    `;
+  }
+} 
+customElements.define('my-app', MyApp);
+```
+
+```html
+<!-- index.html -->
+<!-- this doesn't change -->
+<!-- polyfills, metadata, etc -->
+<script type="module" src="./my-app.js">
+<my-app></my-app>
+```
+
+## Renamed base element export in polymer-element.js {#renamed}
```

Comment:
Is it weird that this section is after the examples section that uses the new syntax?